### PR TITLE
Move RenderTheme* platform files to appropriate locations

### PR DESCRIPTION
**Pull request by annevk**
#### 0b388136aa978b93f94a12000c2a5547fd0c4177
<pre>
Move RenderTheme* platform files to appropriate locations
<a href="https://bugs.webkit.org/show_bug.cgi?id=15756">https://bugs.webkit.org/show_bug.cgi?id=15756</a>

Reviewed by Darin Adler.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/Adwaita.cmake:
* Source/WebCore/platform/SourcesAdwaita.txt:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp: Renamed from Source/WebCore/rendering/RenderThemeAdwaita.cpp.
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h: Renamed from Source/WebCore/rendering/RenderThemeAdwaita.h.
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h: Renamed from Source/WebCore/rendering/RenderThemeCocoa.h.
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm: Renamed from Source/WebCore/rendering/RenderThemeCocoa.mm.
* Source/WebCore/rendering/ios/RenderThemeIOS.h: Renamed from Source/WebCore/rendering/RenderThemeIOS.h.
* Source/WebCore/rendering/ios/RenderThemeIOS.mm: Renamed from Source/WebCore/rendering/RenderThemeIOS.mm.
* Source/WebCore/rendering/mac/RenderThemeMac.h: Renamed from Source/WebCore/rendering/RenderThemeMac.h.
* Source/WebCore/rendering/mac/RenderThemeMac.mm: Renamed from Source/WebCore/rendering/RenderThemeMac.mm.
* Source/WebCore/rendering/playstation/RenderThemePlayStation.cpp: Renamed from Source/WebCore/rendering/RenderThemePlayStation.cpp.
* Source/WebCore/rendering/playstation/RenderThemePlayStation.h: Renamed from Source/WebCore/rendering/RenderThemePlayStation.h.
* Source/WebCore/testing/Internals.cpp:

Only include RenderThemeIOS for IOS_FAMILY platforms.

* Tools/CISupport/build-webkit-org/wkbuild_unittest.py:
(ShouldBuildTest):

Canonical link: <a href="https://commits.webkit.org/273761@main">https://commits.webkit.org/273761@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/7a646d4f7118c1702f45b6f784216993852e007d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36261 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15218 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38478 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38992 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32607 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37488 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17659 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12245 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31275 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36822 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12849 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32169 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11264 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11299 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32366 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40237 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32913 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32735 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37225 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11501 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9381 "Found 1 new test failure: inspector/canvas/resolveContext-2d.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35318 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/36241 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13219 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8293 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11963 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12346 "Built successfully") | | |

### CODE REVIEW

**Comment by annevk** [comment]
@rkirsling there's also RenderThemePlayStation.* but that's not referenced from anywhere so I wasn't sure whether I could rename it without breaking things. Thoughts?

**Comment by rkirsling** [comment]
I believe that should be fine; appreciate you checking!

**Comment by annevk** [comment]
@mcatanzaro can you help with the build failures perhaps? As far as I can tell Headers.cmake is used by the failing platforms to build RenderThemeIOS.h and that file has the correct location. Still somehow they cannot find it when it comes time to deal with includes.

@rkirsling thanks! Once I have clarity on the build failure I'll include a move of the PlayStation files into a playstation directory.

**Comment by lukewarlow** [comment]
There's a reference to some of these files inside of `Tools/CISupport/build-webkit-org/wkbuild_unittest.py` idk if that also needs updating?

File: Source/WebCore/platform/Adwaita.cmake [hunk 1]
```diff
@@ -10,7 +10,7 @@ list(APPEND WebCore_PRIVATE_FRAMEWORK_HEADERS
     platform/adwaita/ScrollbarThemeAdwaita.h
     platform/adwaita/ThemeAdwaita.h
 
-    rendering/RenderThemeAdwaita.h
+    rendering/adwaita/RenderThemeAdwaita.h
```
Comment (philn) [inline Source/WebCore/platform/Adwaita.cmake:13]: Try adding `${WEBCORE_DIR}/rendering/adwaita` in `WebCore_PRIVATE_INCLUDE_DIRECTORIES` ?

Comment (annevk) [reply Source/WebCore/platform/Adwaita.cmake:13]: It seems it was a wrong import in Internals.cpp. However, that is already present which makes me wonder how things are compiling for people locally... Anyway, this should fix it for everyone.

**Comment by mcatanzaro** [comment]
> can you help with the build failures perhaps? As far as I can tell Headers.cmake is used by the failing platforms to build RenderThemeIOS.h and that file has the correct location. Still somehow they cannot find it when it comes time to deal with includes.

I assume you've figured this out, because I see mostly green EWS. :)

**Review by darinadler — APPROVED** [review]

**Pull request merged**